### DISCUSSION TITLE
How to specify isolation level?

### DISCUSSION THREAD

**Discussion by ndc** [linq2db/linq2db#4587] [Q&A] [answered]
Under what isolation level is LinqToDB running? I assume it will use the default isolation level but I'm not sure.

Also, I see that I can specify isolation level when calling BeginTransaction, for example `db.BeginTransaction(System.Data.IsolationLevel.ReadCommitted)`. But is it possible to set it in `LinqToDB.DataOptions`?

**Comment by MaceWindu** [comment, marked as answer]
if you don't specify isolation level explicitly, it is up to provider/database to choose one for you.

Only place where Linq To DB adds isoation level is eager load queries, executed without explicit transaction.

**Reply by ndc** [reply]
Thanks for the info! So to clarify currently there is no place in LinqToDB.DataOptions to specify isolation level?

**Reply by MaceWindu** [reply]
No. Also for eager load too (it is possible to change it, but only through `DataProvider.SqlProviderFlags`)
